File: backend/src/models/AuditLog.ts
import mongoose, { Schema, Document } from 'mongoose';

export interface IAuditLog extends Document {
  adminId: mongoose.Types.ObjectId;
  action: string;
  resource: string;
  resourceId?: string;
  details?: Record<string, any>;
  ipAddress?: string;
  userAgent?: string;
  createdAt: Date;
}

const AuditLogSchema: Schema = new Schema({
  adminId: { type: Schema.Types.ObjectId, ref: 'User', required: true },
  action: { type: String, required: true },
  resource: { type: String, required: true },
  resourceId: { type: String },
  details: { type: Schema.Types.Mixed },
  ipAddress: { type: String },
  userAgent: { type: String },
}, { timestamps: { createdAt: true, updatedAt: false } });

AuditLogSchema.index({ adminId: 1, createdAt: -1 });
AuditLogSchema.index({ resource: 1, resourceId: 1 });
AuditLogSchema.index({ createdAt: -1 });

export default mongoose.model<IAuditLog>('AuditLog', AuditLogSchema);

File: backend/app/models/audit_log.py
from datetime import datetime, timezone
from typing import Optional, Any
from beanie import PydanticObjectId
from pydantic import Field
from app.models.base import BaseDocument


class AuditLog(BaseDocument):
    admin_id: PydanticObjectId = Field(alias="adminId")
    action: str
    resource: str
    resource_id: Optional[str] = Field(default=None, alias="resourceId")
    details: Optional[dict[str, Any]] = None
    ip_address: Optional[str] = Field(default=None, alias="ipAddress")
    user_agent: Optional[str] = Field(default=None, alias="userAgent")
    created_at: datetime = Field(default_factory=lambda: datetime.now(timezone.utc))

    class Settings:
        name = "auditlogs"
        indexes = [
            [("admin_id", -1), ("created_at", -1)],
            [("resource", 1), ("resource_id", 1)],
            [("created_at", -1)],
        ]
